crates/freeze/src/datasets/logs.rs

```diff
@@ -284,5 +297,217 @@ async fn logs_to_df(
+                // and the next batch contains u256s. Might be worth just casting
+                // all as strings
+                Token::Int(i) | Token::Uint(i) => match i.try_into() {
+                    Ok(i) => ints.push(i),
+                    Err(_) => str_ints.push(i.to_string()),
+                },
+                Token::Bool(b) => bools.push(b),
+                Token::String(s) => strings.push(s),
+                Token::Array(_) | Token::FixedArray(_) => {}
+                Token::Tuple(_) => {}
+            }
+        }
+        let mixed_length_err = format!("could not parse column {}, mixed type", name);
+
+        // check each vector, see if it contains any values, if it does, check if it's the same
+        // length as the input data and map to a series
+        if ints.len() > 0 || str_ints.len() > 0 {
+            if str_ints.len() > 0 {
+                str_ints.extend(ints.into_iter().map(|i| i.to_string()));
+                if str_ints.len() != chunk_len {
+                    return Err(mixed_length_err)
+                }
+                return Ok(Series::new(name.as_str(), str_ints))
+            }
+            Ok(Series::new(name.as_str(), ints))
+        } else if bytes.len() > 0 {
+            if bytes.len() != chunk_len {
+                return Err(mixed_length_err)
+            }
+            Ok(Series::new(name.as_str(), bytes))
+        } else if bools.len() > 0 {
+            if bools.len() != chunk_len {
+                return Err(mixed_length_err)
+            }
+            Ok(Series::new(name.as_str(), bools))
+        } else if strings.len() > 0 {
+            if strings.len() != chunk_len {
+                return Err(mixed_length_err)
+            }
+            Ok(Series::new(name.as_str(), strings))
+        } else if addresses.len() > 0 {
+            if addresses.len() != chunk_len {
+                return Err(mixed_length_err)
+            }
+            Ok(Series::new(name.as_str(), addresses))
+        } else {
+            // case where no data was passed
+            Ok(Series::new(name.as_str(), vec![None::<u64>; chunk_len]))
+        }
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use polars::prelude::DataType::Boolean;
+
+    #[test]
+    fn test_mapping_log_into_type_columns() {
+        let raw = "event NewMint(address indexed msgSender, uint256 indexed mintQuantity)";
+        let e = HumanReadableParser::parse_event(raw).unwrap();
+
+        let raw_log = r#"{
+            "address": "0x0000000000000000000000000000000000000000",
+            "topics": [
+                "0x52277f0b4a9b555c5aa96900a13546f972bda413737ec164aac947c87eec6024",
+                "0x00000000000000000000000062a73d9116eda78a78f4cf81602bdc926fb4c0dd",
+                "0x0000000000000000000000000000000000000000000000000000000000000003"
+            ],
+            "data": "0x"
+        }"#;
+
+        let decoder = LogDecoder { raw: raw.to_string(), event: e.clone() };
```

Comment:
use case I have seen coming from analyst, that they would like to get all of the logs (and traces) for a list of contracts on a list of chain, therefor I would change the object oriented approach to a functional one
so you can just pass in a bunch of ABIs and an event(/trace) to decode and get back a JSON/Dict like object at the end

this also could help with a corner case of event collision
RandomEvent(indexed, not indexed)
RandomEvent(indexed, indexed) 
has the same event hash

Reply:
with the way cryo currently does configuration I think supprting your described usecase goes outside the scope of this PR. I don't really see how a functional approach adds much benefit as the context of what needs to be decoded needs to be passed through the call stack as part of the `schema: Table` object in order for decoding to be done correctly. Having that be a struct seems like the right choice to me as it can be extended for more advanced functionality